fastapi/routing.py

```diff
@@ -480,6 +481,7 @@ def __init__(
         route_class: Type[APIRoute] = APIRoute,
         on_startup: Optional[Sequence[Callable[[], Any]]] = None,
         on_shutdown: Optional[Sequence[Callable[[], Any]]] = None,
+        lifespan: Optional[Callable[[Any], AsyncContextManager[Any]]] = None,
```

Comment:
Now it is just inconsistent, no? It's typed `FastAPI` on L74 in applications.py, but here it's typed `Any`, while I would still argue the correct typing would be the `ASGIApp` from `starlette.types`. Because, if you have different orders of middleware (which would all be `ASGIApp`) you don't have to specifically have a `FastAPI` app object in the lifespan param.

Reply:
Interestingly this is actually how it is typed in Starlette I just noticed. But my remark would remain the same, am I missing something maybe?

Reply:
`Any` could be replaced by `FastAPi` which would only be imported if `typing.TYPE_CHECKING` is True (during static analysis and not runtime).